src/App.js

```diff
@@ -1,41 +1,78 @@
 import React from "react";
-import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
+import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import Home from "./pages/Home/";
 import Favorites from "./pages/Favorites/";
 import Search from "./components/Search/";
 import SearchPage from "./pages/SearchPage/";
 import Book from "./pages/Book/";
 import "./App.css";
+import { StyledLink } from "./app.styles";
 
-export default function App() {
-  return (
-    <Router>
-      <div>
-        <nav>
-          <ul>
-            <li>
-              <Link to="/">Home</Link>
-            </li>
-            <li>
-              <Search />
-            </li>
-            <li>
-              <Link to="/favorites">★</Link>
-            </li>
-          </ul>
-        </nav>
+export default class App extends React.Component {
+  state = {
+    favorites: [],
+  };
 
-        <Switch>
-          <Route path="/favorites">
-            <Favorites />
-          </Route>
-          <Route exact path="/">
-            <Home />
-          </Route>
-          <Route component={Book} path="/book/:isbn"></Route>
-          <Route component={SearchPage} path="/search/:searchTerm"></Route>
-        </Switch>
-      </div>
-    </Router>
-  );
+  handleClick = (id) => {
+    const index = this.state.favorites.indexOf(id);
+    const { favorites } = this.state;
+    if (this.state.favorites.includes(id) && index !== -1) {
```

Comment:
why do you destructure favorites and then dont use it?
this goes until line 23...